example/e2e/deploy-federated-deployment.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script has an end to end steps in deploying a federated deployment
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TEST_NS="test-namespace"
+KUBEFED_NS="kube-federation-system"
```

Comment:
Although not necessary but if you are using a variable, you can make it user configurable too.
`KUBEFED_NS="${KUBEFED_NS:-kube-federation-system}"`

Reply:
The goal is not to create an extensible script - it should match the walkthrough exactly.